torch/csrc/jit/codegen/onednn/interface.cpp

```diff
@@ -0,0 +1,145 @@
+namespace onednn {
+
+void fuseGraph(std::shared_ptr<Graph>& g) {
+  // Follow the process of the tensorexpr_fuser in profiling mode:
+  // Remove prim::profile nodes and embed the profile info directly in the
+  // IR in value types to avoid breaking the fusion patterns.
+  // Will add shape guard after LLGA optimization passes and
+  // wipe the tensor type information from the IR, so that it's not
+  // accidentally used by any other pass.
+
+  // We rely on the shape specialization and shape guard to ensure the validity
+  // of the cached compilation in the kernel, thus only support profiling mode.
+  // TODO: add check on oneDNNFusionGroup to ensure allShapesAreKnown on nodes
+  // to fuse: torch/csrc/jit/passes/tensorexpr_fuser.cpp: allShapesAreKnown
+  if (getProfilingMode()) {
+    GRAPH_DUMP(
+        "Before RemoveProfileNodesAndSpecializeTypes. Beginning of LLGA "
+        "optimization pass",
+        g);
+    RemoveProfileNodesAndSpecializeTypes(g);
+    GRAPH_DUMP(
+        "After RemoveProfileNodesAndSpecializeTypes. Before mutation removal",
+        g);
+
+    RemoveTensorMutation(g);
+    RemoveListMutation(g);
+    GRAPH_DUMP("After mutation removal. Before DecomposeOps", g);
+    DecomposeOps(g);
+    GRAPH_DUMP("After DecomposeOps. Before PrepareBinaryForLLGA", g);
+    PrepareBinaryForLLGA(g);
+    GRAPH_DUMP("After PrepareBinaryForLLGA. Before DeferSizeCheck", g);
+    DeferSizeCheck(g);
+    GRAPH_DUMP("After DeferSizeCheck. Before CreateLlgaSubgraphs", g);
+    CreateLlgaSubgraphs(g);
+    GRAPH_DUMP("After CreateLlgaSubgraphs. Before PropagateLayout", g);
+    PropagateLayout(g);
+    GRAPH_DUMP(
+        "After PropagateLayout. Before prepareFusionGroupAndGuardOutputs", g);
+
+    // Add shape guard for profiling mode and wipe the tensor type information
+    // from the IR
+    prepareFusionGroupAndGuardOutputs(g->block());
+    GRAPH_DUMP(
+        "After prepareFusionGroupAndGuardOutputs. Before "
+        "RemoveTensorTypeSpecializations",
+        g);
+    RemoveTensorTypeSpecializations(g);
+    GRAPH_DUMP(
+        "After RemoveTensorTypeSpecializations. End of LLGA optimization pass",
+        g);
+  }
+}
+
+} // namespace onednn
+} // namespace fuser
+
+Operation createLlgaKernel(const Node* node) {
+  auto kernel = std::make_shared<fuser::onednn::LlgaKernel>(node);
+  return [kernel](Stack* stack) {
+    RECORD_FUNCTION(kernel->debugName(), std::vector<c10::IValue>());
+    kernel->run(*stack);
+    return 0;
+  };
+}
+
+RegisterOperators oneDNNFusionGroupOp({
+    torch::jit::Operator(
+        prim::oneDNNFusionGroup,
+        createLlgaKernel,
+        AliasAnalysisKind::INTERNAL_SPECIAL_CASE),
+});
+
+Operation createLlgaGuardKernel(const Node* node) {
+  return [node](Stack* stack) {
+    GRAPH_DEBUG("Guarding node: ", node->kind().toQualString());
+    std::vector<TypePtr> types = node->tys(attr::types);
+    const auto num_inputs = types.size();
+
+    GRAPH_DEBUG("num_inputs to guard: ", num_inputs);
```

Comment:
You probably don't want the DEBUGs running on each invocation, because we will have to do an environment read to check if it's enabled, which will be slow. I would at least guard this with 
```
#if DEBUG
GRAPH_DEBUG
#endif
```
something like that, or check for the environment outside of the op invocation and pass it in so we don't read environment on every run

Reply:
Thanks, this is a very good idea! I'll add this change later today.